bigtable/client/cell.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BIGTABLE_CLIENT_CELL_H_
+#define BIGTABLE_CLIENT_CELL_H_
+
+#include <absl/base/thread_annotations.h>
+#include <absl/strings/string_view.h>
+#include <absl/synchronization/mutex.h>
+#include <bigtable/client/version.h>
+#include <vector>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * The in-memory representation of a Bigtable cell.
+ *
+ * Bigtable stores data in rows, indexes by row keys.  Each row may contain
+ * multiple column families, each column family might contain multiple columns,
+ * and each column has multiple cells indexed by timestamp.  Notice that the
+ * storage is sparse, column families, columns, and timestamps might contain
+ * zero cells.
+ */
+class Cell {
+ public:
+  Cell(absl::string_view row_key, absl::string_view family_name,
```

Comment:
While `absl::string_view` are awesome in some respects their use here means objects of this class become invalidated when the owner of the underlying strings goes away (row?  the streaming class?).  It might be safer to just use strings and copy them, they should not be **that** long.

Reply:
done.

Accessors still return string_view, but I'm not sure at this point what the advantage of this is to simply std::string references. Should I go change them too?